xamarin/Xamarin.Forms
# [iOS] Entry should correctly update Text when typing aid is used

### Description of Change ###

iOS `Entry` does not correctly update its `Text` property when typing aid system is used. This is a limitation of how `EditingChanged` works rather than XF but can be worked around.

Steps to test this PR:

1) Turn on auto-correct
2) Type omw // each character should raise `EditingChanged`
3) Hit return key or tap outside the bounds of `Entry` // does not raise `EditingChanged` event
4) `EditingEnded` works around the issue

1) Turn on auto-correct
2) Type omw // each character should raise `EditingChanged`
3) Hit space key or tap the suggestion // raises `EditingChanged`
4) Default behavior works

### Bugs Fixed ###

- https://bugzilla.xamarin.com/show_bug.cgi?id=31422
- https://bugzilla.xamarin.com/show_bug.cgi?id=42530
- https://bugzilla.xamarin.com/show_bug.cgi?id=38265
